### ISSUE TITLE
Crash while loading project with NoiseTexture2D

### ISSUE THREAD

**Issue by krzmig** [godotengine/godot#66845]
### Godot version

v4.0.beta2.official.f8745f2f7

### System information

Debian 11, Vulkan, GeForce GTX 1050

### Issue description

Hello.
Godot crash when I try to load my project. This only happens on this one project so I am not able to provide a minimal reproduction project.
I found that the problem started in _v4.0-alpha16_, the project starts normal on _v4.0-alpha15_ and earlier.
After removing `*.import` files from the project directory, I am usually able to run the project but only once.

Here is the run log:

```
krzmig@debian:~$ godot4b -v
TextServer: Added interface "Dummy"
TextServer: Added interface "ICU / HarfBuzz / Graphite (Built-in)"
Godot Engine v4.0.beta2.official.f8745f2f7 - https://godotengine.org
XInput: Refreshing devices.
XInput: No touch devices found.
Vulkan devices:
  #0: NVIDIA NVIDIA GeForce GTX 1050 - Supported, Discrete
  #1: Unknown llvmpipe (LLVM 11.0.1, 256 bits) - Unsupported, CPU
Vulkan API 1.2.162 - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce GTX 1050
- Vulkan Variable Rate Shading not supported
- Vulkan multiview supported:
  max view count: 32
  max instances: 134217727
- Vulkan subgroup:
  size: 32
  stages: STAGE_VERTEX, STAGE_TESSELLATION_CONTROL, STAGE_TESSELLATION_EVALUATION, STAGE_GEOMETRY, STAGE_FRAGMENT, STAGE_COMPUTE
  supported ops: FEATURE_BASIC, FEATURE_VOTE, FEATURE_ARITHMETIC, FEATURE_BALLOT, FEATURE_SHUFFLE, FEATURE_SHUFFLE_RELATIVE, FEATURE_CLUSTERED, FEATURE_QUAD, FEATURE_PARTITIONED_NV
  quad operations in all stages
Using present mode: VK_PRESENT_MODE_FIFO_KHR
XcursorGetTheme could not get cursor theme
Using present mode: VK_PRESENT_MODE_FIFO_KHR
Using "default" pen tablet driver...
Creating VMA small objects pool for memory type index 7
Shader 'CanvasSdfShaderRD' SHA256: 60250ee72e612c66c8fbed41e49355dc37f34fe8b4c727559e54172a2cdc308f
(...) Many other shaders...

JoypadLinux: udev enabled and loaded successfully.
PulseAudio: context other
PulseAudio: context other
PulseAudio: context other
PulseAudio: context ready
PulseAudio: Detecting channels for device: alsa_output.pci-0000_28_00.3.analog-stereo
PulseAudio: detected 2 output channels
PulseAudio: audio buffer frames: 512 calculated output latency: 11ms
 
TextServer: Primary interface set to: "ICU / HarfBuzz / Graphite (Built-in)".
CORE API HASH: 2597644884
EDITOR API HASH: 85443156
EditorSettings: Load OK!
Loaded builtin certs
EditorSettings: Save OK!
Editing project: /home/krzmig/Godot/Ballersv4
EditorSettings: Save OK!
Godot Engine v4.0.beta2.official.f8745f2f7 - https://godotengine.org
XR: Clearing primary interface
XR: Removed interfaceNative mobile
XR: Removed interfaceOpenXR
PulseAudio: context terminated
Vulkan API 1.2.162 - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce GTX 1050
Orphan StringName: SceneMultiplayer
StringName: 1 unclaimed string names at exit.
krzmig@debian:~$  
WARNING: FBX file import is enabled in the project settings, but no FBX2glTF path is configured in the editor settings. FBX files will not be imported.
     at: _editor_init (modules/gltf/register_types.cpp:99)
     
WARNING: res://scene/stadium/_component/goal/goal.tscn:4 - ext_resource, invalid UUID: uid://dmb5sl65kuo00 - using text path instead: res://scene/stadium/_component/goal/net.tscn
     at: load (scene/resources/resource_format_text.cpp:448)
(...) Many warnings about the sames for other resources...

ERROR: Condition "!tex" is true.
   at: texture_replace (servers/rendering/renderer_rd/storage_rd/texture_storage.cpp:1223)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.beta2.official (f8745f2f71c79972df66f17a3da75f6e328bc55d)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x38d60) [0x7f85186a2d60] (??:0)
[2] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0x427e4bb] (??:0)
[3] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0x3f73bda] (??:0)
[4] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0x14d2481] (??:0)
[5] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0xe9ba0a] (??:0)
[6] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0x41ecec8] (??:0)
[7] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0x401e9b4] (??:0)
[8] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0x47bd835] (??:0)
[9] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0x4280887] (??:0)
[10] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0x29214d8] (??:0)
[11] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0xe45a46] (??:0)
[12] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0xd8b2f9] (??:0)
[13] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xea) [0x7f851868dd0a] (??:0)
[14] /home/krzmig/Programy/Godot_v4.0-beta2_linux.x86_64() [0xdac6ee] (??:0)
-- END OF BACKTRACE --
================================================================
```

I started this project some time ago, I think on 3.3 (maybe 3.2) but I always used the latest godot release.
I have no more ideas on how to determine what the problem is, and compiling takes too long to search blindly :(
But if someone tells me what to type in code to somehow pinpoint what is causing the error, I can compile the sources.

Best regards

### Steps to reproduce

1. Open a project for editing (whether it's the project manager or directly from the catalog).

**Comment by clayjohn** [comment]
The most helpful thing you could do is build the engine from source using debug flags and then try reproducing the crash again. That way the backtrace should provide more helpful engine. 

In current master you can build with debug using the following command:
```
scons p=linuxbsd target=editor dev_build=yes
```

**Comment by krzmig** [comment]
```
ERROR: Condition "!tex" is true.
   at: texture_replace (servers/rendering/renderer_rd/storage_rd/texture_storage.cpp:1225)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.beta.custom_build (1371a97acfb9263aaca7fa8642e95ba68534fa19)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x38d60) [0x7fcd744c0d60] (??:0)
[2] Object::get_class_name() const (??:0)
[3] Object::emit_signalp(StringName const&, Variant const**, int) (??:0)
[4] Error Object::emit_signal<>(StringName const&) (??:0)
[5] Resource::emit_changed() (??:0)
[6] NoiseTexture2D::_set_texture_image(Ref<Image> const&) (??:0)
[7] NoiseTexture2D::_update_texture() (??:0)
[8] void call_with_variant_args_helper<__UnexistingClass>(__UnexistingClass*, void (__UnexistingClass::*)(), Variant const**, Callable::CallError&, IndexSequence<>) (??:0)
[9] void call_with_variant_args_dv<__UnexistingClass>(__UnexistingClass*, void (__UnexistingClass::*)(), Variant const**, int, Callable::CallError&, Vector<Variant> const&) (??:0)
[10] MethodBindT<>::call(Object*, Variant const**, int, Callable::CallError&) (??:0)
[11] Object::callp(StringName const&, Variant const**, int, Callable::CallError&) (??:0)
[12] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (??:0)
[13] MessageQueue::_call_function(Callable const&, Variant const*, int, bool) (??:0)
[14] MessageQueue::flush() (??:0)
[15] Main::iteration() (??:0)
[16] OS_LinuxBSD::run() (??:0)
[17] /home/krzmig/Programy/Godot/bin/godot.linuxbsd.editor.dev.x86_64(main+0x14e) [0x55f270c8f923] (??:0)
[18] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xea) [0x7fcd744abd0a] (??:0)
[19] /home/krzmig/Programy/Godot/bin/godot.linuxbsd.editor.dev.x86_64(_start+0x2a) [0x55f270c8f71a] (??:0)
-- END OF BACKTRACE --
================================================================
```

Following the class name "NoiseTexture2D", I removed all two noise textures and the project now starts.
I used these textures in the parameters to the shader. I removed both materials and the shader, made them from scratch, adding the noise texture and the project would not open again.
I tried to reproduce this on a clean project but it opens every time...

**Comment by RedMser** [comment]
Not sure if related, but #58818 also reports a crash when loading a project that seems to relate to NoiseTexture2D.

**Comment by clayjohn** [comment]
@RedMser Thank you. Indeed this now appears to be a duplicate of #58818 let's track it there.
